portal-impl/src/com/liferay/portal/internal/servlet/MainServlet.java

```diff
@@ -413,9 +413,8 @@ else if (_log.isInfoEnabled()) {
 					SetupWizardSampleDataUtil.addSampleData(
 						PortalInstances.getDefaultCompanyId());
 				}
-
-				if (PropsValues.
-						SETUP_WIZARD_DEFAULT_ADMIN_PASSWORD_RESET_ENABLED) {
+				else if (PropsValues.
```

Comment:
@dantewang why we still need this checking for SETUP_WIZARD_DEFAULT_ADMIN_PASSWORD_RESET_ENABLED here? You have made SetupWizardSampleDataUtil#addSampleData read this property, isn't that enought?

Tina.

Reply:
> @dantewang why we still need this checking for SETUP_WIZARD_DEFAULT_ADMIN_PASSWORD_RESET_ENABLED here? You have made SetupWizardSampleDataUtil#addSampleData read this property, isn't that enought?
> 
> Tina.

@tinatian, @dantewang, I think the reason is that SETUP_WIZARD_ADD_SAMPLE_DATA and saving Setup Wizard are not related. When configuring Setup Wizard, we only saved SETUP_WIZARD_DEFAULT_ADMIN_PASSWORD_RESET_ENABLED  (SETUP_WIZARD_ADD_SAMPLE_DATA didn't been included) to portal-setup-wizard.properties. If SETUP_WIZARD_ADD_SAMPLE_DATA is true, SetupWizardSampleDataUtil#addSampleData can read the property. If SETUP_WIZARD_ADD_SAMPLE_DATA is false, it needs to check the property SETUP_WIZARD_DEFAULT_ADMIN_PASSWORD_RESET_ENABLED.

By the way, I just did push force.

Reply:
@yuhai I have a new idea.

Maybe we can move the reset password logic to the util, and make `addSampleData` method focusing more on "sample data".

First of all, it seems `SetupWizardSampleDataUtil.addSampleData` is only called by `MainServlet`.

Secondly, if you look at setup wizard, the "Add sample data" checkbox is only checkable when database is changed; and if database is changed, we restart portal, the company and admin user should already take other properties set by setup wizard, like local, email, etc. right? If so, when adding sample data, we don't need to call `updateAdminUser` at all.

If this is agreed, we can remove a lot of parameters related to the admin user from `addSampleData`, and this commit will not be needed.

Could you check this idea?

@tinatian